tests/modules/mosdepth/main.nf

```diff
@@ -2,35 +2,60 @@
 }
 
+workflow test_mosdepth_bed {
+    input  = [
+                [ id:'test', single_end:true ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true) ]
+            ]
+    bed  = [ file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true) ]
+
+    MOSDEPTH ( input, bed, [] )
+}
 
-workflow test_mosdepth_window {
-    input  = [ [ id:'test', single_end:true ],
-               [ file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ],
-               [ file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true) ]
-             ]
-    window = 100
+workflow test_mosdepth_cram {
+    input  = [
+                [ id:'test', single_end:true ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true) ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true) ]
+            ]
+    fasta = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
 
-    MOSDEPTH ( input, [], window )
+    MOSDEPTH ( input, [], fasta )
 }
 
+workflow test_mosdepth_cram_bed {
+    input  = [
+                [ id:'test', single_end:true ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true) ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true) ]
+            ]
+    bed  = [ file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true) ]
+    fasta = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
 
-workflow test_mosdepth_bed {
-    input  = [ [ id:'test', single_end:true ],
-               [ file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ],
-               [ file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true) ]
-             ]
-    bed  = [ file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true) ]
+    MOSDEPTH ( input, bed, fasta )
+}
 
-    MOSDEPTH ( input, bed, [] )
+workflow test_mosdepth_fail {
+    input  = [
+                [ id:'test', single_end:true ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true) ],
+                [ file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true) ]
+            ]
+    bed  = [ file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true) ]
+
+    MOSDEPTH_FAIL ( input, bed, [] )
```

Comment:
Might make sense to add a test for specifying window size and the tests passing too.

Reply:
👍🏻